### PR TITLE
Sample groovy script to add credentials to Jenkins's folder

### PR DESCRIPTION
**Pull request by kuisathaverat**
Sample groovy script to add credentials to Jenkins's folder into global domain

@reviewbybees

### CODE REVIEW

**Comment by imod** [comment]
waiting for @reviewbybees ....

File: scriptler/addCredentialsToFolder.groovy [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ "comment" : "Sample groovy script to add credentials to Jenkins's folder into global domain",
+ "core": "1.609",
+ "authors" : [
+ { name : "Kuisathaverat" }
+ ]
+ } END META**/
+
+import jenkins.model.*
+import com.cloudbees.hudson.plugins.folder.*;
+import com.cloudbees.hudson.plugins.folder.properties.*;
```
Comment (Evildethow) [inline scriptler/addCredentialsToFolder.groovy]: Unused import

File: scriptler/addCredentialsToFolder.groovy [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import com.cloudbees.hudson.plugins.folder.properties.FolderCredentialsProvider.FolderCredentialsProperty;
+import com.cloudbees.plugins.credentials.impl.*;
+import com.cloudbees.plugins.credentials.*;
+import com.cloudbees.plugins.credentials.domains.*;
+
+jenkins = Jenkins.instance
+
+String id = java.util.UUID.randomUUID().toString()
```
Comment (Evildethow) [inline scriptler/addCredentialsToFolder.groovy]: Unnecessary qualified reference.

Can just use `UUID.randomUUID().toString()`

**Comment by Evildethow** [comment]
The loop can be re-written to:

```
jenkins.getAllItems(Folder.class)
    .findAll { it.name == 'FolderName' }
    .each {
  AbstractFolder<?> folderAbs = AbstractFolder.class.cast(it)
  FolderCredentialsProperty property = folderAbs.getProperties().get(FolderCredentialsProperty.class)
  property.getStore().addCredentials(Domain.global(), c)
  println property.getCredentials().toString()
}
```

If you want to be more functional (and groovy, if that floats your boat) :)

Just a comment.

File: scriptler/addCredentialsToFolder.groovy [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+Credentials c = new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL, id, "description:"+id, "user", "password")
+
+for (folder in jenkins.getAllItems(Folder.class)) {
+  if(folder.name.equals('FolderName')){
+	AbstractFolder<?> folderAbs = AbstractFolder.class.cast(folder)
+    FolderCredentialsProperty property = folderAbs.getProperties().get(FolderCredentialsProperty.class)
+    property.getStore().addCredentials(Domain.global(), c)
```
Comment (Evildethow) [inline scriptler/addCredentialsToFolder.groovy]: `property` needs a `null` check?

`.get(FolderCredentialsProperty.class)` can return `null` according to the source:

```
    public <U extends T> U get(Class<U> type) {
        for (T t : data)
            if(type.isInstance(t))
                return type.cast(t);
        return null;
    }
```

**Comment by Evildethow** [comment]
🐛  for the missing `null` check. Happy to reverse this if you know that it will never be null. 

All my other comments are cosmetic changes, only if you want to do.

File: scriptler/addCredentialsToFolder.groovy [hunk 4]
```diff
@@ -0,0 +1,30 @@
+/*** BEGIN META {
+ "name" : "add credentials to folder",
+ "comment" : "Sample groovy script to add credentials to Jenkins's folder into global domain",
+ "core": "1.609",
+ "authors" : [
+ { name : "Kuisathaverat" }
+ ]
+ } END META**/
+
+import jenkins.model.*
```
Comment (Evildethow) [inline scriptler/addCredentialsToFolder.groovy]: `jenkins.model.*` is pre-imported.

**Comment by kuisathaverat** [comment]
@Evildethow I added the null check and the others changes , Could you check if all is ok now?

**Comment by Evildethow** [comment]
@kuisathaverat LGTM 🐝

**Pull request merged**